make-ffmpeg.py

```diff
@@ -4,152 +4,175 @@
 import os
 import sys
 import subprocess
-import schedulelib
 import argparse
-import shlex
-from PIL import ImageFont
+import ssl
 from configparser import ConfigParser
-import json
+from pathlib import PurePath
 import platform
-import ssl
+
+from PIL import ImageFont
+import schedulelib
 ssl._create_default_https_context = ssl._create_unverified_context
 
-# Parse arguments
-parser = argparse.ArgumentParser(
-    description='C3VOC Intro-Outro-Generator - Variant which renders only using video filters in ffmpeg',
-    usage="./make-ffmpeg.py yourproject/",
-    formatter_class=argparse.RawTextHelpFormatter)
+FRAME_WIDTH = 1920
+
+
+class TextConfig:
+    inpoint: float
+    outpoint: float
+    x: int
+    y: int
+
+    fontfile_path: str
+    fontfamily: str
+
+    fontsize: int
+    fontcolor: str
+    bordercolor: str = None  # border is added, if a color is set
+
+    def uses_fontfile(self):
+        return self.fontfile_path is not None
+
+    def parse(self, cparser_sect, default_fontfile, default_fontfamily, default_fontcolor):
+        self.inpoint = cparser_sect.getfloat('in')
+        self.outpoint = cparser_sect.getfloat('out')
+        self.x = cparser_sect.getint('x')
+        self.y = cparser_sect.getint('y')
+
+        self.fontcolor = cparser_sect.get('fontcolor', default_fontcolor)
+
+        fontfile = cparser_sect.get('fontfile', default_fontfile)
+        fontfamily = cparser_sect.get('fontfamily', default_fontfamily)
+        if fontfile != None and fontfamily is None:
+            self.fontfile_path = str(PurePath(args.project, fontfile).as_posix())
+
+            if not os.path.exists(self.fontfile_path):
+                error("Font file {} in Project Path is missing".format(self.fontfile_path))
+
+        elif fontfamily != None and fontfile is None:
+            self.fontfamily = fontfamily
+
+        else:
+            error("Either provide a 'fontfamily' or 'fontfile', not both")
+
+        self.fontsize = cparser_sect.getint('fontsize')
+        self.bordercolor = cparser_sect.get('bordercolor', None)
+
+    def fit_text(self, text: str):
+        font = ImageFont.truetype(
+            self.fontfile_path, size=self.fontsize, encoding="unic")
```

Comment:
This never worked when a `fontfamily` is used. How can we make it work of do we just remove the option to use a font family instead of an explicit file?

Reply:
I just removed the feature as it's broken and supposedly not used